src/main/scala/microsites/github/GitHubOps.scala

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright 2017-2020 47 Degrees, LLC. <http://www.47deg.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package microsites.github
+
+import java.io.File
+
+import cats.data.{NonEmptyList, OptionT}
+import cats.effect.{ConcurrentEffect, IO, Sync, Timer}
+import cats.implicits._
+import com.github.marklister.base64.Base64._
+import github4s.Github
+import github4s.GithubResponses._
+import github4s.domain._
+import microsites.Exceptions._
+import microsites.github.Config._
+import microsites.ioops.{FileReader, IOUtils => FIO}
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+class GitHubOps[F[_]: ConcurrentEffect: Timer](
+    owner: String,
+    repo: String,
+    accessToken: Option[String],
+    fileReader: FileReader = FileReader
+)(implicit ec: ExecutionContext) {
+
+  val gh: Github[F]                = Github[F](accessToken)
+  val headers: Map[String, String] = Map("user-agent" -> "sbt-microsites")
+
+  def commitDir(branch: String, message: String, dir: File): F[Ref] =
+    Sync[F]
+      .delay(fileReader.fetchDirsRecursively(List(dir)))
+      .handleError(e => throw IOException(s"Error fetching files recursively", Some(e)))
```

Comment:
what's the reason for all the throws? :slightly_smiling_face:

Reply:
Just wanted to manage the potential IO exception, by mapping it to an `IOException` (re-raising it to abort the execution)